webapp/src/main/java/com/epam/esm/security/ApplicationSecurity.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Bean
+    public CustomAuthorizationFilter authorizationFilter(ObjectMapper objectMapper) {
+        authorizationFilter = new CustomAuthorizationFilter(objectMapper);
+        return authorizationFilter;
+    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(userService).passwordEncoder(passwordEncoder);
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable();
+        http.sessionManagement().sessionCreationPolicy(STATELESS);
+        configureCertificateAccessRules(http);
+        configureTagAccessRules(http);
+        configureUserAccessRules(http);
+        http.authorizeRequests().anyRequest().authenticated();
+        http.addFilter(authenticationFilter);
+        http.addFilterBefore(authorizationFilter, CustomAuthenticationFilter.class);
+        http.exceptionHandling()
+//                .authenticationEntryPoint(exceptionEntryPoint)
+                .accessDeniedHandler(accessDeniedHandler);
+    }
+
+    @Bean
+    public AuthenticationManager authenticationManager() throws Exception {
+        return super.authenticationManager();
+    }
+
+    private void configureCertificateAccessRules(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .antMatchers(GET, SERVLET_PATH + "/certificates").permitAll()
+                .antMatchers(GET, SERVLET_PATH + "/certificates/**").permitAll()
+                .antMatchers(POST, SERVLET_PATH + "/certificates").hasRole(ADMIN)
+                .antMatchers(PATCH, SERVLET_PATH + "/certificates/**").hasRole(ADMIN)
+                .antMatchers(DELETE, SERVLET_PATH + "/certificates/**").hasRole(ADMIN);
+    }
+
+    private void configureTagAccessRules(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .antMatchers(GET, SERVLET_PATH + "/tags").permitAll()
+                .antMatchers(GET, SERVLET_PATH + "/most_popular_tag").permitAll()
+                .antMatchers(GET, SERVLET_PATH + "/tags/**").permitAll()
+                .antMatchers(POST, SERVLET_PATH + "/tags").hasRole(ADMIN)
+                .antMatchers(DELETE, SERVLET_PATH + "/tags/**").hasRole(ADMIN);
+    }
+
+    private void configureUserAccessRules(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .antMatchers(POST, SERVLET_PATH + "/users").anonymous()
+                .antMatchers(GET, SERVLET_PATH + "/users").hasRole(ADMIN)
+                .antMatchers(GET, SERVLET_PATH + "/users/{id}").access("@applicationSecurity.checkUserPermission(authentication, #id)")
+                .antMatchers(GET, SERVLET_PATH + "/users/{id}/**").access("@applicationSecurity.checkUserPermission(authentication, #id)")
+                .antMatchers(POST, SERVLET_PATH + "/users/{id}/orders").access("@applicationSecurity.checkUserPermission(authentication, #id)");
+    }
```

Comment:
If you change the url, you will have to go through all the methods and replace it, which will lead to errors and poor performance.

Please, use SPeL 
For example:
![image](https://user-images.githubusercontent.com/49719217/141324864-f27871aa-a41f-4a31-9323-4138409f4114.png)